wseagar/eight-dollars
# Fix/classes retainment

Fix for a small bug introduced in #46.

When a node gets mutated (rather than added), the code would set the check mark classname to `undefined` instead of the right one.

Fixed by retaining the original class of the node in `data-eight-dollars-original-classes` attribute.

Test case:
- Go to [@growing_daniel](https://twitter.com/growing_daniel)
- Scroll down to the retweet from [@sandofsky](https://twitter.com/sandofsky)
- Navigate to [@sandofsky's](https://twitter.com/sandofsky) profile
- Check that his checkmark is displayed correctly.

**Before**:
<img width="612" alt="image" src="https://user-images.githubusercontent.com/2031472/201839647-3fddafed-193c-46c1-b440-f99ccac55edb.png">
<img width="607" alt="image" src="https://user-images.githubusercontent.com/2031472/201839685-59b2f973-d044-4429-8fbb-a85aa77e6490.png">
<img width="606" alt="image" src="https://user-images.githubusercontent.com/2031472/201839712-63b29151-76a6-4f00-8c6e-ef7e0c6fad42.png">

**After**:
<img width="608" alt="image" src="https://user-images.githubusercontent.com/2031472/201839817-fa3678ab-a380-4294-84fa-007b717b4b01.png">
<img width="605" alt="image" src="https://user-images.githubusercontent.com/2031472/201839856-959fccab-d11b-4763-967e-cefb4ae11526.png">
<img width="605" alt="image" src="https://user-images.githubusercontent.com/2031472/201839887-5e68ba5e-e7f9-4eb5-850e-e604dea30bb3.png">

**Bug scope**:
The bug is only affecting "Enable Text" mode. The icon-only mode worked fine. I don't think anyone reported this yet.
